src/java.desktop/share/classes/javax/swing/RepaintManager.java

```diff
@@ -390,11 +390,8 @@ public synchronized void removeInvalidComponent(JComponent component) {
             delegate.removeInvalidComponent(component);
             return;
         }
-        if(invalidComponents != null) {
-            int index = invalidComponents.indexOf(component);
-            if(index != -1) {
-                invalidComponents.remove(index);
-            }
+        if (invalidComponents != null) {
+            invalidComponents.remove(component);
```

Comment:
I guess the `addInvalidComponent` above can be simplified as well?

Reply:
Simplified how?

Reply:
the loop over invalidComponents

Reply:
Do you suggest to use `List.indexOf` instead of manual loop?
https://github.com/openjdk/jdk/blob/375ac6d446332f0763ce294b200143ff63865cf6/src/java.desktop/share/classes/javax/swing/RepaintManager.java#L366-L371

This loop uses `==` comparison, while `List.indexOf` uses `equals()`. It could change semantic

Reply:
ok, the it is fine as is.

Reply:
Yet there could be a logical error anyway… `addInvalidComponent` uses identity comparison when adding a component but `removeInvalidComponent` relies on `equals` method to remove a component.

If a component overrides `equals`, it could lead to unpredictable behaviour: a wrong component could be removed.

Reply:
Then it's better to use "==" in all cases, what do you think?

Reply:
Thank you, Andrey.

P.S. The link is broken, it doesn't lead to the JBS issue.